src/subscription_manager/managercli.py

```diff
@@ -500,6 +532,20 @@ def main(self, args=None):
 
         # do the work, catch most common errors here:
         try:
+            if self.proxy_hostname or \
```

Comment:
Could this be inside the 
``` python
if self.require_connection():
``` 
block? 

I think this could end up testing the proxy connection even on cli commands that don't need networking. Having
a self.proxy_hostname etc implies self.require_connection() == True, but potentially not for the config change cases.